### PR TITLE
use containerd 1.1.5, enable w/ non-vhd flows

### LINKED ISSUE #184: "distro": "ubuntu" + non-avail hyperkube broken
Was testing on a containerd-configured cluster, so this should also be tested against moby:

In the provision logs:

```
+ extractHyperkube img
+ CLI_TOOL=img
+ path=/home/hyperkube-downloads/1.10.12
+ mkdir -p /home/hyperkube-downloads/1.10.12
+ pullContainerImage img k8s.gcr.io/hyperkube-amd64:v1.10.12
+ CLI_TOOL=img
+ DOCKER_IMAGE_URL=k8s.gcr.io/hyperkube-amd64:v1.10.12
+ retrycmd_if_failure 60 1 1200 img pull k8s.gcr.io/hyperkube-amd64:v1.10.12
+ retries=60
+ wait_sleep=1
+ timeout=1200
+ shift
+ shift
+ shift
++ seq 1 60
+ for i in '$(seq 1 $retries)'
+ timeout 1200 img pull k8s.gcr.io/hyperkube-amd64:v1.10.12
Pulling k8s.gcr.io/hyperkube-amd64:v1.10.12...
Pulled: sha256:f23bcbac5c5c0d1c8f5302c5fcddbaf55a684be3a7fdf1281cad403f976b9fcd
Size: 174.7MiB
+ '[' 0 -eq 0 ']'
+ break
+ echo Executed '"img' pull 'k8s.gcr.io/hyperkube-amd64:v1.10.12"' 1 times
Executed "img pull k8s.gcr.io/hyperkube-amd64:v1.10.12" 1 times
+ [[ img == \d\o\c\k\e\r ]]
+ img unpack -o /home/hyperkube-downloads/1.10.12 k8s.gcr.io/hyperkube-amd64:v1.10.12
destination directory already exists: /home/hyperkube-downloads/1.10.12
+ [[ UBUNTU == COREOS ]]
+ cp /home/hyperkube-downloads/1.10.12/hyperkube /usr/local/bin/kubelet-1.10.12
cp: cannot stat '/home/hyperkube-downloads/1.10.12/hyperkube': No such file or directory
+ mv /home/hyperkube-downloads/1.10.12/hyperkube /usr/local/bin/kubectl-1.10.12
mv: cannot stat '/home/hyperkube-downloads/1.10.12/hyperkube': No such file or directory
+ mv /usr/local/bin/kubelet-1.10.12 /usr/local/bin/kubelet
mv: cannot stat '/usr/local/bin/kubelet-1.10.12': No such file or directory
+ mv /usr/local/bin/kubectl-1.10.12 /usr/local/bin/kubectl
mv: cannot stat '/usr/local/bin/kubectl-1.10.12': No such file or directory
+ chmod a+x /usr/local/bin/kubelet /usr/local/bin/kubectl
chmod: cannot access '/usr/local/bin/kubelet': No such file or directory
chmod: cannot access '/usr/local/bin/kubectl': No such file or directory
```

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Update containerd to 1.1.5:

https://github.com/containerd/containerd/releases/tag/v1.1.5

We should implement containerd 1.2 for k8s 1.11 and above at some point, as it appears to be the preferred (and breaking for k8s.1.10 and below) containerd version for modern k8s.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 

fixes #184

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
use containerd 1.1.5, enable w/ non-vhd flows
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
Tested out O.K.:

```
$ kubectl get nodes -o json
2018/12/19 13:44:17 NAME                        STATUS    ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
k8s-agentpool1-31042654-0   Ready     agent     1m        v1.11.6   <none>        Ubuntu 16.04.5 LTS   4.15.0-1035-azure   containerd://1.2.1
k8s-agentpool1-31042654-1   Ready     agent     1m        v1.11.6   <none>        Ubuntu 16.04.5 LTS   4.15.0-1035-azure   containerd://1.2.1
k8s-agentpool1-31042654-2   Ready     agent     1m        v1.11.6   <none>        Ubuntu 16.04.5 LTS   4.15.0-1035-azure   containerd://1.2.1
k8s-master-31042654-0       Ready     master    1m        v1.11.6   <none>        Ubuntu 16.04.5 LTS   4.15.0-1035-azure   containerd://1.2.1
```

File: parts/k8s/kubernetesinstalls.sh [hunk 1]
```diff
@@ -197,9 +200,9 @@ installImg() {
 extractHyperkube() {
     CLI_TOOL=$1
     path="/home/hyperkube-downloads/${KUBERNETES_VERSION}"
-    mkdir -p "$path"
     pullContainerImage $CLI_TOOL ${HYPERKUBE_URL}
     if [[ "$CLI_TOOL" == "docker" ]]; then
+        mkdir -p "$path"
```
Comment (jackfrancis) [inline parts/k8s/kubernetesinstalls.sh]: Pre-creating the directory works for the docker extraction implementation (`docker run ...` below), but it was breaking the non-docker extraction implementation (`img unpack...` below). This change fixes #184.

File: parts/k8s/kubernetesinstalls.sh [hunk 2]
```diff
@@ -174,19 +183,13 @@ installAzureCNI() {
 }
 
 installContainerd() {
-    containerd --version
```
Comment (jackfrancis) [inline parts/k8s/kubernetesinstalls.sh]: Decomposed this to resemble the Azure CNI implementation

File: parts/k8s/kubernetesinstalls.sh [hunk 3]
```diff
@@ -5,6 +5,10 @@ CC_SOCKET_IN_TMP=/opt/azure/containers/cc-proxy.socket.in
 CNI_CONFIG_DIR="/etc/cni/net.d"
 CNI_BIN_DIR="/opt/cni/bin"
 CNI_DOWNLOADS_DIR="/opt/cni/downloads"
+CONTAINERD_DOWNLOADS_DIR="/opt/containerd/downloads"
+CRI_CONTAINERD_VERSION="1.1.5"
+CONTAINERD_DOWNLOAD_URL="${CONTAINERD_DOWNLOAD_URL_BASE}cri-containerd-${CRI_CONTAINERD_VERSION}.linux-amd64.tar.gz"
+CONTAINERD_TGZ_TMP=$(echo ${CONTAINERD_DOWNLOAD_URL} | cut -d "/" -f 5)
```
Comment (jackfrancis) [inline parts/k8s/kubernetesinstalls.sh]: All of this re-organization is to accommodate `CONTAINERD_TGZ_TMP` re-use across the download and install functions.

File: packer/install-dependencies.sh [hunk 4]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+CONTAINERD_DOWNLOAD_URL_BASE="https://storage.googleapis.com/cri-containerd-release/"
```
Comment (jackfrancis) [inline packer/install-dependencies.sh]: The general re-purposing of `CONTAINERD_TGZ_TMP ` (which needs `CONTAINERD_DOWNLOAD_URL`, and thus `CONTAINERD_DOWNLOAD_URL_BASE`) in the CSE install script means that we have to pass in a value for `CONTAINERD_DOWNLOAD_URL_BASE` before we source the script.

**Comment by jackfrancis** [comment]
@mboersma @tariq1890 tagging this PR as `v0.29.0`. This is ready for master but shouldn't be included in the `v0.28` patch release cycle.

File: parts/k8s/kubernetesinstalls.sh [hunk 5]
```diff
@@ -174,19 +183,13 @@ installAzureCNI() {
...
-        retrycmd_get_tarball 120 5 "$CONTAINERD_TGZ_TMP" "$CONTAINERD_DOWNLOAD_URL" || exit $ERR_CONTAINERD_DOWNLOAD_TIMEOUT
-        tar -xzf "$CONTAINERD_TGZ_TMP" -C /
-        rm -f "$CONTAINERD_TGZ_TMP"
-        sed -i '/\[Service\]/a ExecStartPost=\/sbin\/iptables -P FORWARD ACCEPT' /etc/systemd/system/containerd.service
-        echo "Successfully installed cri-containerd..."
+    if [[ ! -f "$CONTAINERD_DOWNLOADS_DIR/${CONTAINERD_TGZ_TMP}" ]]; then
+        downloadContainerd
     fi
+    tar -xzf "$CONTAINERD_DOWNLOADS_DIR/$CONTAINERD_TGZ_TMP" -C /
+    rm -Rf $CONTAINERD_DOWNLOADS_DIR &
```
Comment (mboersma) [inline parts/k8s/kubernetesinstalls.sh]: Always capitalize that `R` please folks.

**Pull request merged**